test/e2e_env/kubernetes/gateway/cross-mesh.go

```diff
@@ -0,0 +1,178 @@
+
+		Expect(setup.Setup(env.Cluster)).To(Succeed())
+	}
+
+	// Before each test, install the gateway and routes.
+	JustBeforeEach(func() {
+		Expect(k8s.KubectlApplyFromStringE(
+			env.Cluster.GetTesting(),
+			env.Cluster.GetKubectlOptions(gatewayTestNamespace), fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: MeshGateway
+metadata:
+  name: %s
+mesh: %s
+spec:
+  selectors:
+  - match:
+      kuma.io/service: %s
+  conf:
+    listeners:
+    - port: %d
+      protocol: HTTP
+      crossMesh: true
+      hostname: gateway.mesh
+      tags:
+        hostname: gateway.mesh
+`, edgeGateway, gatewayMesh, edgeGateway, gatewayPort)),
+		).To(Succeed())
+
+		Expect(
+			env.Cluster.GetKumactlOptions().KumactlList("meshgateways", gatewayMesh),
+		).To(ContainElement(edgeGateway))
+
+		Expect(k8s.KubectlApplyFromStringE(
+			env.Cluster.GetTesting(),
+			env.Cluster.GetKubectlOptions(gatewayTestNamespace), fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: MeshGatewayRoute
+metadata:
+  name: %s
+mesh: %s
+spec:
+  selectors:
+  - match:
+      kuma.io/service: %s
+  conf:
+    http:
+      rules:
+      - matches:
+        - path:
+            match: PREFIX
+            value: /
+        backends:
+        - destination:
+            kuma.io/service: %s_%s_svc_80 # Matches the echo-server we deployed.
+`, edgeGateway, gatewayMesh, edgeGateway, echoServer, gatewayTestNamespace)),
+		).To(Succeed())
+
+		Expect(
+			env.Cluster.GetKumactlOptions().KumactlList("meshgatewayroutes", gatewayMesh),
+		).To(ContainElement(edgeGateway))
+	})
+
+	// Before each test, install the GatewayInstance to provision
+	// dataplanes. Note that we expose the gateway inside the cluster
+	// with a ClusterIP service. This makes it easier for the tests
+	// to figure out the IP address to send requests to, since the
+	// alternatives are a load balancer (we don't have one) or node port
+	// (would need to inspect nodes).
+	JustBeforeEach(func() {
+		Expect(k8s.KubectlApplyFromStringE(
+			env.Cluster.GetTesting(),
+			env.Cluster.GetKubectlOptions(gatewayTestNamespace), fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: MeshGatewayInstance
+metadata:
+  name: %s
+  annotations:
+    kuma.io/mesh: %s
+spec:
+  replicas: 1
+  serviceType: ClusterIP
+  tags:
+    kuma.io/service: %s
+`, edgeGateway, gatewayMesh, edgeGateway)),
+		).To(Succeed())
+	})
+
+	// Before each test, verify that we have the Dataplanes that we expect to need.
+	JustBeforeEach(func() {
+		Expect(env.Cluster.VerifyKuma()).To(Succeed())
+
+		// Synchronize on the dataplanes coming up.
+		Eventually(func(g Gomega) {
+			dataplanes, err := env.Cluster.GetKumactlOptions().KumactlList("dataplanes", gatewayMesh)
+			g.Expect(err).ToNot(HaveOccurred())
+			// Dataplane names are generated, so we check for a partial match.
+			g.Expect(dataplanes).Should(ContainElement(ContainSubstring(echoServer)))
+			g.Expect(dataplanes).Should(ContainElement(ContainSubstring(edgeGateway)))
+		}, "60s", "1s").Should(Succeed())
```

Comment:
I also find those checks unnecessary, we don't do this in other tests.
We have `Eventually` on proxyRequestToGateway, so even if the dps are not up yet, they eventually will be.